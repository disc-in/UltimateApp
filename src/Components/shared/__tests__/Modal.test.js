import React from 'react';
import { render } from '@testing-library/react-native';
import { Text } from 'react-native';

import Modal from '../Modal';

beforeEach(() => jest.useFakeTimers()); // for Modal behaviour

describe('<Modal />', () => {
  it('renders correctly', () => {
    const onClose = jest.fn();
    const { toJSON } = render(
      <Modal visible title="My Title" onClose={onClose}>
        <Text>Hello</Text>
      </Modal>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
