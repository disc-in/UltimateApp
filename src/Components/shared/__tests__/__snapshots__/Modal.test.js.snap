// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  useNativeDriver={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": "10%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 10,
          "maxHeight": "80%",
          "maxWidth": "80%",
          "minWidth": "60%",
          "padding": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        My Title
      </Text>
      <Text>
        Hello
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 120,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
